sharding/data_modeling.rst

```diff
@@ -24,12 +24,77 @@ If your situation resembles either of these cases then the next step is to decid
 
 If you're migrating an existing database to the Citus multi-tenant architecture then some of your tables may lack a column for the application-specific tenant id. You will need to add one and fill it with the correct values. This will denormalize your tables slightly. For more details and a concrete example of backfilling the tenant id, see our guide to :ref:`Multi-Tenant Migration <transitioning_mt>`.
 
+Typical Multi-Tenant Scenario
+-----------------------------
+
+Most SaaS applications already have the notion of tenancy built into their data model. In the following, we will look at an example schema from the online advertising space. In this example, a web advertising platform has tenants that it refers to as accounts. Each account holds and tracks advertising clicks across various campaigns.
+
+.. code-block:: postgres
+
+  CREATE TABLE accounts (
+    id integer PRIMARY KEY,
+    name text NOT NULL,
+    image_url text NOT NULL
+  );
+
+  CREATE TABLE ads (
+    id integer NOT NULL,
+    account_id integer NOT NULL,
+    campaign_id integer NOT NULL,
+    name text NOT NULL,
+    image_url text NOT NULL,
+    target_url text NOT NULL,
+    impressions_count bigint DEFAULT 0 NOT NULL,
+    clicks_count bigint DEFAULT 0 NOT NULL,
+
+    PRIMARY KEY (account_id, id),
+    FOREIGN KEY (account_id) REFERENCES accounts,
+  );
+
+  CREATE TABLE clicks (
+    id uuid DEFAULT gen_random_uuid() NOT NULL,
+    account_id integer NOT NULL,
+    ad_id integer NOT NULL,
+    clicked_at timestamp without time zone NOT NULL,
+    site_url text NOT NULL,
+    cost_per_click_usd numeric(20,10),
+    user_ip inet NOT NULL,
+    user_data jsonb NOT NULL,
+
+    PRIMARY KEY (account_id, id),
+    FOREIGN KEY (account_id) REFERENCES accounts,
+    FOREIGN KEY (account_id, ad_id) REFERENCES ads (account_id, id)
+  );
+
+  SELECT create_distributed_table('accounts',  'id');
+  SELECT create_distributed_table('ads',       'account_id');
+  SELECT create_distributed_table('clicks',    'account_id');
+
+Citus provides full SQL coverage for queries that include a tenant id. It supports foreign key constraints as long as they include the tenant id column (see for instance the two foreign keys in the :code:`clicks` table above). Full-featured queries involving joins, grouping or aggregate functions work without modification as long as they filter and join by tenant id. For instance here is how to count newly arriving clicks per campaign for account id=9700:
```

Comment:
I think the previous table schema looked good in that we started showing a tree structure.

This ending still looks weak to me for two reasons.

* We show an example, but don't spend enough time talking about primary and foreign keys and requirements around making them work.
* We also show an example ``SELECT`` query and conclude the section. In #291, we're showing more features around Citus and its benefits. Could we look to incorporate a few examples to this section?
* In our multi-tenant "lecture" at PGConf SV, we also showed an example for an update that touched multiple shards. Do we want to show or provide a link to that here?